### ISSUE TITLE
[Bug]: Video streams are not starting in specific stream quality

### ISSUE THREAD

**Issue by Dean-Corso** [vzhd1701/gridplayer#78]
### GridPlayer version

v0.4.3

### What OS are you using?

Windows

### OS Version / Linux distribution

Microsoft Windows 10 Pro

### Bug description

Hi vzhd1701,

in some cases I try to run any live stream or also video stream (On-Demand / Not live / Youtube) and the video/stream is not starting and runs into Buffering Timeout. 

Example: I found this YT video URL which is not starting for me [Video](https://www.youtube.com/watch?v=9pQtXLQY51A) when using the default settings for Video/Streaming Videos/Stream Quality: Best. In this case the resolver YT-dlp is used (I can not change this to Streamlink etc) and trying to load the video with Best parameter. When it fails then I can switch the stream quality in context menu to a lower one 360p or 144p which then works. I think its a YT-dlp problem so it also happens when using YT-DL CLI version itsef. Below the file informations for this YT-URL
```
ID  EXT   RESOLUTION FPS │   FILESIZE   TBR PROTO │ VCODEC        VBR ACODEC      ABR ASR MORE INFO
───────────────────────────────────────────────────────────────────────────────────────────────────────────
sb2 mhtml 48x27        0 │                  mhtml │ images                                storyboard
sb1 mhtml 80x45        0 │                  mhtml │ images                                storyboard
sb0 mhtml 160x90       0 │                  mhtml │ images                                storyboard
249 webm  audio only     │    8.81MiB   47k https │ audio only        opus        47k 48k low, webm_dash
139 m4a   audio only     │    9.06MiB   49k https │ audio only        mp4a.40.5   49k 22k low, m4a_dash
250 webm  audio only     │   11.62MiB   63k https │ audio only        opus        63k 48k low, webm_dash
251 webm  audio only     │   23.69MiB  128k https │ audio only        opus       128k 48k medium, webm_dash
140 m4a   audio only     │   24.04MiB  129k https │ audio only        mp4a.40.2  129k 44k medium, m4a_dash
17  3gp   176x144      6 │   14.65MiB   79k https │ mp4v.20.3     79k mp4a.40.2    0k 22k 144p
160 mp4   256x144     25 │   13.83MiB   74k https │ avc1.4d400c   74k video only          144p, mp4_dash
278 webm  256x144     25 │   15.32MiB   83k https │ vp9           83k video only          144p, webm_dash
242 webm  426x240     25 │   24.97MiB  134k https │ vp9          134k video only          240p, webm_dash
133 mp4   426x240     25 │   28.47MiB  153k https │ avc1.4d4015  153k video only          240p, mp4_dash
243 webm  640x360     25 │   42.43MiB  229k https │ vp9          229k video only          360p, webm_dash
134 mp4   640x360     25 │   51.86MiB  279k https │ avc1.4d401e  279k video only          360p, mp4_dash
244 webm  854x480     25 │   66.82MiB  360k https │ vp9          360k video only          480p, webm_dash
18  mp4   640x360     25 │   85.80MiB  462k https │ avc1.42001E  462k mp4a.40.2    0k 44k 360p
135 mp4   854x480     25 │  116.61MiB  628k https │ avc1.4d401e  628k video only          480p, mp4_dash
247 webm  1280x720    25 │  123.43MiB  665k https │ vp9          665k video only          720p, webm_dash
22  mp4   1280x720    25 │ ~129.05MiB  679k https │ avc1.64001F  679k mp4a.40.2    0k 44k 720p
248 webm  1920x1080   25 │  217.27MiB 1170k https │ vp9         1170k video only          1080p, webm_dash
136 mp4   1280x720    25 │  227.38MiB 1225k https │ avc1.64001f 1225k video only          720p, mp4_dash
137 mp4   1920x1080   25 │  436.82MiB 2353k https │ avc1.640028 2353k video only          1080p, mp4_dash
```
The best marked video (value: 22) fails and the other 2 streams V+A (value: 18 & 17) are working to play. I also added the errors of YT-dlp below into log excerpt window. I was also using the latest version YT-dlp...
https://github.com/yt-dlp/yt-dlp/releases/tag/2022.07.18
...not whether its really a bug of yt-dlp because its working mostly fine to play all 3 V+A streams but in some cases like this not. Maybe you can check this out what the problem is. Thank you.

### Log excerpt

```shell
2022-07-20 22:53:26,808 (7864) | EventWaiter | ERROR | Waiting for buffering timed out after 30 seconds
2022-07-20 22:53:26,810 (7864) | PlayerProcessSingleVLCHW | ERROR | Buffering timeout

Error Info from YT-DL:
--------------------------------------------------------------------------
[debug] File locking is not supported. Proceeding without locking
[download] Destination: -
←[K[download] ←[0;94m  0.6%←[0m of 125.99MiB at ←[0;32m   3.27MiB/s←[0m ETA ←[0;33m00:38←[0m[download] Got server HTTP error: Downloaded 746924 bytes, expected 132105947 bytes. Retrying (attempt 1 of 10) ...


←[0;31mERROR:←[0m Did not get any data blocks
  File "yt_dlp\__main__.py", line 17, in <module>
  File "yt_dlp\__init__.py", line 940, in main
  File "yt_dlp\__init__.py", line 932, in _real_main
  File "yt_dlp\YoutubeDL.py", line 3281, in download
  File "yt_dlp\YoutubeDL.py", line 3257, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1422, in extract_info
  File "yt_dlp\YoutubeDL.py", line 1431, in wrapper
  File "yt_dlp\YoutubeDL.py", line 1521, in __extract_info
  File "yt_dlp\YoutubeDL.py", line 1578, in process_ie_result
  File "yt_dlp\YoutubeDL.py", line 2661, in process_video_result
  File "yt_dlp\YoutubeDL.py", line 3160, in process_info
  File "yt_dlp\YoutubeDL.py", line 2860, in dl
  File "yt_dlp\downloader\common.py", line 444, in download
  File "yt_dlp\downloader\http.py", line 376, in real_download
  File "yt_dlp\downloader\http.py", line 345, in download
  File "yt_dlp\YoutubeDL.py", line 971, in report_error
  File "yt_dlp\YoutubeDL.py", line 903, in trouble
```

**Comment by vzhd1701** [comment]
Can't reproduce. Works OK for me with portable.

Do you have a custom yt-dlp config in your system? It should be at `C:\Users\%USERNAME%\AppData\Roaming\yt-dlp`, if not then search for anything with `yt-dlp` in your User profile directory.

**Comment by Dean-Corso** [comment]
So I have checked the same URL again today and now its working. Hhhmm. Strange. So the last two days it was not working to play the BEST stream option with YT-dlp and now it works. No I don't use and have any config on system for yt-dlp or anything. I'am just using raw direct commandlines when using CLI tools. I also got that problem in the past few times with YT or CB and a specific stream quality URL is not working as I told before. Must be a special case and YT-dlp will fail. Maybe just a char / special char issue to read the stream URL wrong or something like that or whatever. No problem of your tool itself. Maybe I can get some more infos when it happens again next time.
